### ISSUE TITLE
kafka error in jaeger collector 

### ISSUE THREAD

**Issue by spyvip** [jaegertracing/jaeger#2373]
**Describe the bug**
````
{"level":"error","ts":1596586452.7701614,"caller":"kafka/writer.go:58","msg":"kafka server: Tried to send a message to a replica that is not the leader for some partition. Your metadata is out of date.","stacktrace":"github.com/jaegertracing/jaeger/plugin/storage/kafka.NewSpanWriter.func2\n\tgithub.com/jaegertracing/jaeger/plugin/storage/kafka/writer.go:58"}
{"level":"error","ts":1596586452.7701683,"caller":"kafka/writer.go:58","msg":"kafka server: Tried to send a message to a replica that is not the leader for some partition. Your metadata is out of date.","stacktrace":"github.com/jaegertracing/jaeger/plugin/storage/kafka.NewSpanWriter.func2\n\tgithub.com/jaegertracing/jaeger/plugin/storage/kafka/writer.go:58"}
{"level":"error","ts":1596586452.7701766,"caller":"kafka/writer.go:58","msg":"kafka server: Tried to send a message to a replica that is not the leader for some partition. Your metadata is out of date.","stacktrace":"github.com/jaegertracing/jaeger/plugin/storage/kafka.NewSpanWriter.func2\n\tgithub.com/jaegertracing/jaeger/plugin/storage/kafka/writer.go:58"}
{"level":"error","ts":1596586452.7701836,"caller":"kafka/writer.go:58","msg":"kafka server: Tried to send a message to a replica that is not the leader for some partition. Your metadata is out of date.","stacktrace":"github.com/jaegertracing/jaeger/plugin/storage/kafka.NewSpanWriter.func2\n\tgithub.com/jaegertracing/jaeger/plugin/storage/kafka/writer.go:58"}
{"level":"error","ts":1596586452.7701962,"caller":"kafka/writer.go:58","msg":"kafka server: Tried to send a message to a replica that is not the leader for some partition. Your metadata is out of date.","stacktrace":"github.com/jaegertracing/jaeger/plugin/storage/kafka.NewSpanWriter.func2\n\tgithub.com/jaegertracing/jaeger/plugin/storage/kafka/writer.go:58"}
````

**To Reproduce**
1. 
2.
3.

**Expected behavior**

**Screenshots**

**Version (please complete the following information):**
 - OS: [e.g. Linux]
 - Jaeger version: [e.g. 1.8]
 - Deployment: [e.g. bare metal, Docker, Kubernetes, ...]

**What troubleshooting steps did you try?**
Try to follow https://www.jaegertracing.io/docs/latest/troubleshooting/ and describe how far you were able to progress and/or which steps did not work.

**Additional context**

**Comment by kevinearls** [comment]
Hi @spyvip Can you give us some more information about what you were doing when you saw this problem?   What platform are you running on, what versions of Jaeger and Kafka are you using, and the yaml you used to create your Jaeger instance?  Thanks.

**Comment by spyvip** [comment]
@kevinearls 
when kafka rebalance
jaeger 1.18

**Comment by kevinearls** [comment]
Hi @spyvip  Thanks for the update, but it would be helpful for us to have more information in order to reproduce this.  Could you give us some more information about what you are doing, including:

-- Are you running on kubernetes or openshift?  If so what version
-- What version of Kafka are you using and how are you running it?
-- What version of Jaeger are you running?

Thanks!

**Comment by spyvip** [comment]
@kevinearls 

-- Are you running on kubernetes or openshift? If so what version
 // aliyun k8s
````
系统镜像 : CentOS Linux 7 (Core) | 内核版本 : 3.10.0-693.2.2.el7.x86_64
-- | --
Kubelet 版本 : v1.12.6-aliyun.1 | Kube-Proxy 版本 : v1.12.6-aliyun.1
docker://17.6.2
````

 
-- What version of Kafka are you using and how are you running it?
// docker.io/bitnami/kafka:2.3.0-debian-9-r4

-- What version of Jaeger are you running?
// 1.18

-- How are you deploying Jaeger? Are you using an operator?
// k8s


-- If you are deploying via the operator can you send us the Jaeger CR? Otherwise can you tell us what options you are using when running the collector and ingested?
![image](https://user-images.githubusercontent.com/1212932/90306885-ecdca300-df03-11ea-9341-2dba6d09499f.png)

**Comment by objectiser** [comment]
@spyvip I believe there was a bug in kafka 2.3.0 (fixed in 2.3.1) related to rebalancing. Could you try with a more recent version?

**Comment by objectiser** [comment]
I'll close this issue - but if upgrading the kafka version doesn't resolve the issue, please reopen.
